getting_started/scripting/c_sharp/c_sharp_basics.rst

```diff
@@ -3,93 +3,122 @@
-A good starting point for checking its capabilities is the `Compatibility <http://www.mono-project.com/docs/about-mono/compatibility/>`_ page in the Mono documentation.
+.. warning:: C# support is a new feature available since Godot 3.0.
+             As such, you may still run into some issues, or find spots
+             where the documentation could be improved.
+             Please report issues with C# in Godot on the
+             `engine GitHub page <https://github.com/godotengine/godot/issues>`_,
+             and any documentation issues on the
+             `documentation GitHub page <https://github.com/godotengine/godot-docs/issues>`_.
+
+This page provides a brief introduction to C#, both what it is and
+how to use it in Godot. Afterwards, you may want to look at
+:ref:`how to use specific features <doc_c_sharp_features>`, read about the
+:ref:`differences between the C# and the GDScript API <doc_c_sharp_differences>`
+and (re)visit the :ref:`Scripting section <doc_scripting>` of the
+step-by-step tutorial.
+
+C# is a high-level programming language developed by Microsoft. In Godot,
+it is implemented with the Mono 5.x .NET framework, including full support
+for C# 7.0. Mono is an open source implementation of Microsoft's .NET Framework
+based on the ECMA standards for C# and the Common Language Runtime.
+A good starting point for checking its capabilities is the
+`Compatibility <http://www.mono-project.com/docs/about-mono/compatibility/>`_
+page in the Mono documentation.
 
 .. note:: This is **not** a full-scale tutorial on the C# language as a whole.
         If you aren't already familiar with its syntax or features,
-        see the `Microsoft C# guide <https://docs.microsoft.com/en-us/dotnet/csharp/index>`_ or look for a suitable introduction elsewhere.
-
-Setup C# for Godot
-------------------
-
-Download and install the `Mono <http://www.mono-project.com/download/>`_ SDK.
-
-If you are using Godot 3.0.2, you must use Mono 5.4.
-
-Godot 3.0.3+ requires Mono 5.12 on all platforms.
+        see the
+        `Microsoft C# guide <https://docs.microsoft.com/en-us/dotnet/csharp/index>`_
+        or look for a suitable introduction elsewhere.
 
-.. note:: To download Mono v5.12+ on a Mac, locate the "Stable Channel" link from the `Mono Downloads Page <http://www.mono-project.com/download/>`_. The Visual Studio channel is an earlier version of Mono and will not work with Godot 3.0.3+.
+Setting up C# for Godot
+-----------------------
 
-You also need MSBuild (at least version 15.0), which should come with the Mono installation.
+Download and install the latest version of the
+`Mono SDK <http://www.mono-project.com/download/>`_. As of Godot 3.1 beta 3,
+the version number doesn't matter since Godot bundles its own Mono 5.18
+installation. We only need the Mono installation for NuGet and MSBuild
+which are required to use C# in Godot.
```

Comment:
Mono's MSBuild gives issues on Windows. We recently changed the default build tool on Windows to Visual Studio's MSBuild. The user needs to either install Visual Studio 2017 or `Build Tools for Visual Studio 2017 ` if they don't want the entire IDE (https://visualstudio.microsoft.com/downloads/).
The plan is to bundle MSBuild as well but that will take some time.